### PR TITLE
Feature/basic exercises4

### PR DESCRIPTION
**Pull request by awwkoala**

### CODE REVIEW

File: src/main/java/com/github/awwkoala/learning/object/Factorial.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package com.github.awwkoala.learning.object;
+
+public class Factorial {
+  //  Napisz metodę, która obliczy silnię liczby naturalnej, podanej jako argument. Metoda powinna być rekurencyjna.
+  public double factorial(double a) {
+    double result;
+    if (a < 0) {
+      throw new IllegalArgumentException("The number cannot be less than 0.");
+    } else if (a <= 1) {
+      return result = 1;
```
Comment (lgonczarik) [inline src/main/java/com/github/awwkoala/learning/object/Factorial.java]: What is this?

File: src/main/java/com/github/awwkoala/learning/object/Factorial.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    }
+    result = a * factorial(a - 1);
+    return result;
```
Comment (lgonczarik) [inline src/main/java/com/github/awwkoala/learning/object/Factorial.java]: just return.

File: src/main/java/com/github/awwkoala/learning/object/Factorial.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package com.github.awwkoala.learning.object;
+
+public class Factorial {
+  //  Napisz metodę, która obliczy silnię liczby naturalnej, podanej jako argument. Metoda powinna być rekurencyjna.
+  public double factorial(double a) {
+    double result;
```
Comment (lgonczarik) [inline src/main/java/com/github/awwkoala/learning/object/Factorial.java]: You dont need this.

File: src/main/java/com/github/awwkoala/learning/object/Human.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  //  Utwórz klasę Human reprezentującą człowieka, musi posiadać atrybuty takie jak wiek, waga, wzrost, imię i płeć.
+//  Klasa powinna także zawierać metody getAge, getWeight, getHeight, getName, isMale.
+  private int age;
+  private int weight;
+  private int height;
+  private String name;
+  private boolean male;
+
+  public Human(int age, int weight, int height, String name, boolean male) {
+    if (age <= 0 || weight <= 0 || height <= 0) {
```
Comment (lgonczarik) [inline src/main/java/com/github/awwkoala/learning/object/Human.java]: Human age cannot be 0?

File: src/test/java/com/github/awwkoala/learning/figure/CircleTest.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
+package com.github.awwkoala.learning.figure;
+
+import org.assertj.core.api.Assertions;
+import org.testng.annotations.Test;
+
+public class CircleTest {
+
+  private double a = 3;
```
Comment (lgonczarik) [inline src/test/java/com/github/awwkoala/learning/figure/CircleTest.java]: Constants :)

File: src/test/java/com/github/awwkoala/learning/figure/CircleTest.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+  private double b = 0;
+  private double c = -3;
+  private Circle circlePos = new Circle(a);
```
Comment (lgonczarik) [inline src/test/java/com/github/awwkoala/learning/figure/CircleTest.java]: move this to method

File: src/test/java/com/github/awwkoala/learning/figure/RectangleTest.java [hunk 7]
```diff
@@ -0,0 +1,51 @@
+package com.github.awwkoala.learning.figure;
+
+import org.assertj.core.api.Assertions;
+import org.testng.annotations.Test;
+
+public class RectangleTest {
+  private double a = 2;
```
Comment (lgonczarik) [inline src/test/java/com/github/awwkoala/learning/figure/RectangleTest.java]: const

File: src/test/java/com/github/awwkoala/learning/figure/RectangleTest.java [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+  private double b = 3;
+  private double c = 0;
+  private double d = -2;
+  private Rectangle rectanglePos = new Rectangle(a, b);
```
Comment (lgonczarik) [inline src/test/java/com/github/awwkoala/learning/figure/RectangleTest.java]: move this to method

File: src/test/java/com/github/awwkoala/learning/figure/TriangleTest.java [hunk 9]
```diff
@@ -0,0 +1,52 @@
+package com.github.awwkoala.learning.figure;
+
+import org.assertj.core.api.Assertions;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.*;
+
+public class TriangleTest {
+  private double a = 2;
```
Comment (lgonczarik) [inline src/test/java/com/github/awwkoala/learning/figure/TriangleTest.java]: const.

File: src/test/java/com/github/awwkoala/learning/figure/TriangleTest.java [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+  private double b = 3;
+  private double c = 0;
+  private double d = -2;
+  private Triangle trianglePos = new Triangle(a,b);
```
Comment (lgonczarik) [inline src/test/java/com/github/awwkoala/learning/figure/TriangleTest.java]: move this to method

File: src/test/java/com/github/awwkoala/learning/object/FactorialTest.java [hunk 11]
```diff
@@ -0,0 +1,31 @@
+package com.github.awwkoala.learning.object;
+
+import org.assertj.core.api.Assertions;
+import org.testng.annotations.Test;
+
+public class FactorialTest {
+  private double numPos = 5;
```
Comment (lgonczarik) [inline src/test/java/com/github/awwkoala/learning/object/FactorialTest.java]: const

File: src/test/java/com/github/awwkoala/learning/object/FactorialTest.java [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+  private double numZero = 0;
+  private double numNeg = -5;
+  Factorial fac = new Factorial();
```
Comment (lgonczarik) [inline src/test/java/com/github/awwkoala/learning/object/FactorialTest.java]: to method?

File: src/test/java/com/github/awwkoala/learning/object/MyNumberTest.java [hunk 13]
```diff
@@ -0,0 +1,69 @@
+package com.github.awwkoala.learning.object;
+
+import org.assertj.core.api.Assertions;
+import org.testng.annotations.Test;
+
+public class MyNumberTest {
+  private double oddNum = 3;
```
Comment (lgonczarik) [inline src/test/java/com/github/awwkoala/learning/object/MyNumberTest.java]: consts?

File: src/test/java/com/github/awwkoala/learning/object/MyNumberTest.java [hunk 14]
```diff
@@ -0,0 +1,69 @@
...
+  private double evenNum = 4;
+  private double zero = 0;
+  private MyNumber myNumOdd = new MyNumber(oddNum);
```
Comment (lgonczarik) [inline src/test/java/com/github/awwkoala/learning/object/MyNumberTest.java]: method?

File: src/test/java/com/github/awwkoala/learning/object/SecondRectangleTest.java [hunk 15]
```diff
@@ -0,0 +1,21 @@
+package com.github.awwkoala.learning.object;
+
+import com.github.awwkoala.learning.figure.Rectangle;
+import org.assertj.core.api.Assertions;
+import org.testng.annotations.Test;
+
+public class SecondRectangleTest {
+  private double a = 2;
```
Comment (lgonczarik) [inline src/test/java/com/github/awwkoala/learning/object/SecondRectangleTest.java]: const

File: src/test/java/com/github/awwkoala/learning/object/SecondRectangleTest.java [hunk 16]
```diff
@@ -0,0 +1,21 @@
...
+  private double b = 3;
+  private SecondRectangle rectanglePos = new SecondRectangle(a, b);
```
Comment (lgonczarik) [inline src/test/java/com/github/awwkoala/learning/object/SecondRectangleTest.java]: method

File: src/main/java/com/github/awwkoala/learning/object/Human.java [hunk 17]
```diff
@@ -0,0 +1,43 @@
...
+public class Human {
+  //  Utwórz klasę Human reprezentującą człowieka, musi posiadać atrybuty takie jak wiek, waga, wzrost, imię i płeć.
+//  Klasa powinna także zawierać metody getAge, getWeight, getHeight, getName, isMale.
+  private int age;
+  private int weight;
+  private int height;
+  private String name;
+  private boolean male;
+
+  public Human(int age, int weight, int height, String name, boolean male) {
```
Comment (ilusi0npl) [inline src/main/java/com/github/awwkoala/learning/object/Human.java:15]: Please use lombok for getters and setters

**Review by lgonczarik — APPROVED** [review]

**Review by lgonczarik — APPROVED** [review]

**Pull request merged**